src/eachobs.jl

```diff
@@ -255,3 +255,41 @@ end
     e.parallel && throw(ArgumentError("Transducer fold protocol not supported on parallel data loads"))
     _dataloader_foldl1(rf, val, e, ObsView(e.data))
 end
+
+# Base uses this function for composable array printing, e.g. adjoint(view(::Matrix)))
+function Base.showarg(io::IO, e::DataLoader, toplevel)
+    print(io, "DataLoader(")
+    Base.showarg(io, e.data, false)
```

Comment:
The intention with overloading this (not just `show`) is that things like `CuIterator(DataLoader(adjoint(::Matrix{...` could then work without packages knowing about each other.

Reply:
I agree, but I thought there would be a default implementation for `repeat` that calls the iterator repeatedly?

Reply:
Maybe there should be. I thought it was only arrays, but turns out to repeat strings too:
```
julia> methods(repeat)
# 6 methods for generic function "repeat" from Base:
 [1] repeat(A::AbstractArray; inner, outer)
     @ abstractarraymath.jl:392
 [2] repeat(A::AbstractArray, counts...)
     @ abstractarraymath.jl:355
 [3] repeat(c::Char, r::Integer)
     @ strings/string.jl:349
 [4] repeat(c::AbstractChar, r::Integer)
     @ strings/string.jl:348
 [5] repeat(s::Union{SubString{String}, String}, r::Integer)
     @ strings/substring.jl:251
 [6] repeat(s::AbstractString, r::Integer)
     @ strings/basic.jl:715
```
I guess it's eager for all of those, so perhaps unclear whether `repeat((x for x in 1:3), 3)` should make an iterator or collect.